pkg/rpc/client.go

```diff
@@ -128,3 +124,92 @@ func (c *Client) rpcRequest(ctx context.Context, data io.Reader) ([]byte, error)
 
 	return body, nil
 }
+
+func (c *Client) getResponse(ctx context.Context, method string, params []interface{}, result HasRPCError) error {
+	body, err := c.rpcRequest(ctx, formatRPCRequest(method, params))
+	// check if there was an error making the request:
+	if err != nil {
+		return fmt.Errorf("%s RPC call failed: %w", method, err)
+	}
+	// log response:
+	klog.V(2).Infof("%s response: %v", method, string(body))
+
+	// unmarshal the response into the predicted format
+	if err = json.Unmarshal(body, result); err != nil {
+		return fmt.Errorf("failed to decode %s response body: %w", method, err)
+	}
+
+	if result.getError().Code != 0 {
+		return fmt.Errorf("RPC error: %d %v", result.getError().Code, result.getError().Message)
+	}
+
+	return nil
+}
+
+func (c *Client) GetEpochInfo(ctx context.Context, commitment Commitment) (*EpochInfo, error) {
+	var resp response[EpochInfo]
+	if err := c.getResponse(ctx, "getEpochInfo", []interface{}{commitment}, &resp); err != nil {
+		return nil, err
+	}
+	return &resp.Result, nil
+}
+
+func (c *Client) GetVoteAccounts(ctx context.Context, params []interface{}) (*VoteAccounts, error) {
+	var resp response[VoteAccounts]
+	if err := c.getResponse(ctx, "getVoteAccounts", params, &resp); err != nil {
+		return nil, err
+	}
+	return &resp.Result, nil
+}
+
+func (c *Client) GetVersion(ctx context.Context) (string, error) {
+	var resp response[struct {
+		Version string `json:"solana-core"`
+	}]
+	if err := c.getResponse(ctx, "getVersion", []interface{}{}, &resp); err != nil {
+		return "", err
+	}
+	return resp.Result.Version, nil
+}
+
+func (c *Client) GetSlot(ctx context.Context) (int64, error) {
+	var resp response[int64]
+	if err := c.getResponse(ctx, "getSlot", []interface{}{}, &resp); err != nil {
+		return 0, err
+	}
+	return resp.Result, nil
+}
+
+func (c *Client) GetBlockProduction(ctx context.Context, firstSlot *int64, lastSlot *int64) (*BlockProduction, error) {
+	// format params:
+	params := make([]interface{}, 1)
+	if firstSlot != nil {
+		params[0] = map[string]interface{}{
```

Comment:
Why interface? Just to make it easier? The [docs](https://solana.com/docs/rpc/http/getblockproduction#result) don't look too bad.

Http json apis are 100x easier in python :)

Reply:
Not sure, just used what was already in the code - will definitely like to clean up further

Reply:
Fair enough, we can do that at a later date.